### PR TITLE
move from 3.6 to 3.7

### PR DESCRIPTION
**Pull request by andre-merzky**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
         python-version: [ '3.7', '3.8', '3.9', '3.10', '3.11' ]
         include:
           - os: ubuntu-20.04
-            python-version: '3.6'
+            python-version: '3.7'
```
Comment (mtitov) [inline .github/workflows/ci.yml:58]: these 3 lines could be removed then (3.7 is in matrix already)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -97,7 +97,7 @@ jobs:
...
```
Comment (mtitov) [inline .github/workflows/ci.yml:100]: same as above, all three lines to be removed

File: setup.py [hunk 3]
```diff
@@ -167,9 +167,9 @@ def get_version(_mod_root):
 
 
 # ------------------------------------------------------------------------------
-# check python version, should be >= 3.6
+# check python version, should be >= 3.7
 if sys.hexversion < 0x03060000:
```
Comment (mtitov) [inline setup.py:171]: `if sys.hexversion < 0x03070000:`

**Review by mtitov — COMMENTED** [review]
have a second thought on this change - should it be merged not with the upcoming release? (or even the one after the next?). While SAGA is still the main interface to HPCs, thus not to rush with this enforcement

File: setup.py (hunk 3 above)
Comment (andre-merzky) [reply setup.py:171]: Ah, I knew we had this somewhere - overlooked...  Thanks!

File: .github/workflows/ci.yml (hunk 2 above)
Comment (andre-merzky) [reply .github/workflows/ci.yml:100]: will do

**Comment by andre-merzky** [comment]
> have a second thought on this change - should it be merged not with the upcoming release? (or even the one after the next?). While SAGA is still the main interface to HPCs, thus not to rush with this enforcement

Yeah, I don't mind pushing that down the road a bit.

**Comment by eirrgang** [comment]
Note that you will also need to update the `.readthedocs.yml` or docs builds will fail.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (andre-merzky) [reply .github/workflows/ci.yml:94]: @mtitov : Sorry, I lost the context of this comment - what lines should be removed?

**Comment by mtitov** [comment]
@andre-merzky we can close this PR considering that it is part of other one, which we push right now, and I'll add comments there in regards of this topic.

**Comment by andre-merzky** [comment]
> @andre-merzky we can close this PR considering that it is part of other one, which we push right now, and I'll add comments there in regards of this topic.

Agree - thanks!

**Pull request closed without merging**